File: Topic/018_equalizeHist/018_equalizeHist/018_equalizeHist.cpp
#include <opencv2/opencv.hpp>
#include <iostream>

using namespace std;
using namespace cv;

int main(int argc, char** argv) {
	Mat src = imread("H:\\OpenCV_Learning\\test.jpg");
	if (src.empty()) {
		printf("could not load image...\n");
		return -1;
	}
	namedWindow("input", WINDOW_AUTOSIZE);
	Mat gray, dst;
	cvtColor(src, gray, COLOR_BGR2GRAY);
	imshow("input", gray);
	equalizeHist(gray, dst);
	imshow("eq", dst);

	waitKey(0);
	return 0;

}

File: Topic/018_equalizeHist/018_equalizeHist.py
import cv2 as cv
import numpy as np
from matplotlib import pyplot as plt


src = cv.imread("H:\\OpenCV_Learning\\test.jpg")
gray = cv.cvtColor(src, cv.COLOR_BGR2GRAY)
cv.namedWindow("input", cv.WINDOW_AUTOSIZE)
cv.imshow("input", gray)
dst = cv.equalizeHist(gray)
cv.imshow("eq", dst)
cv.waitKey(0)
cv.destroyAllWindows()